### PR TITLE
Add vault name screen

### LINKED ISSUE #180: Add "Name your vault" screen
Add "Name your vault" screen before starting Keygen

### PR DESCRIPTION
**Pull request by aminsato**
Fix #180

### CODE REVIEW

File: app/src/main/java/com/vultisig/wallet/presenter/keygen/KeygenFlowViewModel.kt [hunk 1]
```diff
@@ -85,22 +85,9 @@ internal class KeygenFlowViewModel @Inject constructor(
...
     suspend fun setData(vaultId: String, context: Context) {
         // start mediator server
-        val allVaults = vaultRepository.getAll()
 
-        val vault = if (vaultId == Screen.KeygenFlow.DEFAULT_NEW_VAULT) {
-            var newVaultName = ""
-            var idx = 1
-            while (true) {
-                newVaultName = "New vault ${allVaults.size + idx}"
-                if (allVaults.find { it.name == newVaultName } == null) {
```
Comment (johnnyluo) [inline app/src/main/java/com/vultisig/wallet/presenter/keygen/KeygenFlowViewModel.kt:95]: Can you please keep these logic there , so the vault name can be optional for user?

File: app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    val viewModel = hiltViewModel<NamingVaultViewModel>()
+    val uiModel by viewModel.uiModel.collectAsState()
+    val snackBarHostState = remember { SnackbarHostState() }
+    LaunchedEffect(key1 = Unit) {
+        viewModel.snackBarChannel.collect { message ->
+            message?.let {
+                snackBarHostState.showSnackbar(message)
+            }
+        }
+    }
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt:61]: I don't see any reason to duplicate this code. There's VaultRenameScreen which is basically the same

**Review by YGolybev — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/vultisig/wallet/presenter/keygen/KeygenFlowViewModel.kt (hunk 1 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/presenter/keygen/KeygenFlowViewModel.kt]: Done

File: app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt (hunk 2 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt]: Done

Comment (YGolybev) [reply app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt]: I don't see any improvements about it, NamingVaultScreen still seems to be a duplicate of VaultRenameScreen

Comment (Ehsan-saradar) [reply app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt]: @YGolybev 
only ui of two screens are same. but the arguments of destinations and the functionality of the screens differ from each other. if i merge them into one screen the functionality of that will lost its single responsibility and make its viewmodel more complex and unclear.

Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt]: @YGolybev 

only ui of two screens are same. but the arguments of destinations and the functionality of the screens differ from each other. if i merge them into one screen the functionality of that will lost its single responsibility and make its viewmodel more complex and unclear.

Comment (YGolybev) [reply app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt]: You can reuse UI with different ViewModels. There's even 

```
private fun VaultRenameScreen(
    navHostController: NavHostController,
    uiModel: VaultEditUiModel,
    snackBarHostState: SnackbarHostState,
    onSave: () -> Unit = {},
    onChangeName: (name: String) -> Unit = {},
)
```

for that case already.

File: app/src/main/java/com/vultisig/wallet/presenter/vault_setting/vault_edit/VaultRenameScreen.kt [hunk 3]
```diff
@@ -1,204 +1,39 @@
...
-    val context = LocalContext.current
 
     LaunchedEffect(key1 = Unit) {
         viewModel.loadData()
-        viewModel.channelFlow.collect { event ->
-            when (event) {
-                is ShowSnackBar ->
-                    snackBarHostState.showSnackbar(event.message.asString(context))
-            }
-        }
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/presenter/vault_setting/vault_edit/VaultRenameScreen.kt:70]: Did that part of the code get lost or become irrelevant?

Comment (Ehsan-saradar) [reply app/src/main/java/com/vultisig/wallet/presenter/vault_setting/vault_edit/VaultRenameScreen.kt]: Fixed

Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/presenter/vault_setting/vault_edit/VaultRenameScreen.kt]: Check again please

File: app/src/main/java/com/vultisig/wallet/ui/models/VaultAccountsViewModel.kt [hunk 4]
```diff
@@ -42,51 +41,78 @@ internal data class AccountUiModel(
...
     private var loadAccountsJob: Job? = null
 
     fun loadData(vaultId: String) {
+        this.vaultId = vaultId
         loadVaultName(vaultId)
-        loadAccounts(vaultId)
+        loadAccounts(
+            vaultId = vaultId,
+            showRefreshing = false,
+        )
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/ui/models/VaultAccountsViewModel.kt:65]: There's changes from my other PR in this one, please do something with git history of this branch

**Review by YGolybev — APPROVED** [review]

**Comment by aminsato** [comment]
Plase take a look @YGolybev

**Pull request merged**